flytekit/core/context_manager.py

```diff
@@ -180,6 +180,9 @@ def __init__(self, prefix: str, task_resolver: Optional["TaskResolverMixin"] = N
         self._branch_nodes: List[Node] = []
         self._task_resolver = task_resolver
 
+        # Count of entities as they are compiled {entity.name: count}
+        self._entity_counts: Dict[str, int] = {}
```

Comment:
Can we rename to `_entity_name_counts` instead or something?  `_entity_counts` makes it seem like we're storing the pointer location of the object as the key, rather than just the name.